browser/notifications/notification_platform_bridge_brave_custom_notification.cc

```diff
@@ -52,12 +55,14 @@ class PassThroughDelegate : public brave_custom_notification::NotificationDelega
 
   void Close(bool by_user) override {
     brave_ads::AdsNotificationHandler* handler = new brave_ads::AdsNotificationHandler(static_cast<content::BrowserContext*>(profile_));
```

Comment:
Please see `NotificationDisplayServiceImpl::AddNotificationHandler` on how to change this to use a smart pointer